File: DisasterServer/Entities/ExellerClone.cs
using DisasterServer.Maps;
using DisasterServer.Session;
using DisasterServer.State;

namespace DisasterServer.Entities;

public class ExellerClone : Entity
{
	public byte ID;

	public ushort PID;

	public sbyte Dir;

	public ExellerClone(ushort pid, byte id, int x, int y, sbyte dir)
	{
		Dir = dir;
		PID = pid;
		ID = id;
		X = x;
		Y = y;
	}

	public override TcpPacket? Destroy(Server server, Game game, Map map)
	{
		return new TcpPacket(PacketType.SERVER_EXELLERCLONE_STATE, (byte)1, ID);
	}

	public override TcpPacket? Spawn(Server server, Game game, Map map)
	{
		return new TcpPacket(PacketType.SERVER_EXELLERCLONE_STATE, (byte)0, ID, PID, (ushort)X, (ushort)Y, Dir);
	}

	public override UdpPacket? Tick(Server server, Game game, Map map)
	{
		return null;
	}
}

File: disasterserver/Entities/ExellerClone.cpp
#include "ExellerClone.hpp"

#include "Packet.hpp"

using namespace DisasterServer;
using namespace DisasterServer::Entities;

ExellerClone::ExellerClone(entityId id, Server &server, GameState &state, const Vector2 &position, int8_t dir, clientId owner) :
    Entity(id, server, state, "exclone", position), dir(dir), owner(owner) {}

ExellerClone::~ExellerClone() = default;

bool ExellerClone::init() {
    Packet pack(PacketType::SERVER_EXELLERCLONE_STATE);
    pack.write<uint8_t>(0);
    pack.write<entityId>(id);
    pack.write<clientId>(owner);
    pack.writeVector2(position);
    pack.write<int8_t>(dir);
    pack.sendBroadcast(server);
    return true;
}

bool ExellerClone::uninit() {
    return true;
}
